src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -34,12 +34,7 @@ internal struct SNIErrorDetails
     // and surfacing objects to the user.
     internal sealed partial class TdsParser
     {
-        static TdsParser()
-        {
-            // For CoreCLR, we need to register the ANSI Code Page encoding provider before attempting to get an Encoding from a CodePage
-            Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
```

Comment:
you need to remove the line 
```
<Reference Include="System.Text.Encoding.CodePages" />
```
from System.Data.SqlClient.csproj

other than that LGTM